lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -25,7 +25,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
   serialize: function(_model) {
     var model = _model || this.context,
         names = this.names,
-        handler = this.handler;
+        handler = this.handler,
+        serializer = this.serializer || (handler && handler.serialize);
```

Comment:
@MiguelMadero the scenario you are envisioning is correct. In order to properly support lazy-loading it will be a pre-requisite that developers use the new approach. It is something that we will likely add to the guides for using Engines and lazy-loading.

However, we can not simply drop the old approach as this would break existing applications and not follow semver.